cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -127,6 +128,14 @@
   List<RunRecord> getRuns(Id.Program id, ProgramRunStatus status, long startTime, long endTime, int limit);
 
   /**
+   * Fetches the run records for the particular status.
+   * @param status  status of the program running/completed/failed or all
+   * @param limit   max number of entries to fetch for this history call
+   * @return        lis of logged runs
+   */
+  List<RunRecord> getRuns(ProgramRunStatus status, int limit);
```

Comment:
Currently it is only called from the `DistributedProgramRuntimeService` and limit is set to 100. We can add predicate as a parameter, which would filter based on the list of twillRunId. Will add TODO for this and do cleanup in another PR, if thats ok.